source/types/types.go

```diff
@@ -1,10 +1,11 @@
 package srctypes
 
 const (
-	DockerImageScheme = "docker-image"
-	GitScheme         = "git"
-	LocalScheme       = "local"
-	HTTPScheme        = "http"
-	HTTPSScheme       = "https"
-	OCIScheme         = "oci-layout"
+	DockerImageScheme     = "docker-image"
+	DockerImageBlobScheme = "docker-image-blob"
```

Comment:
Surfacing from Slack:

> I think that docker-image-blob:// feels too generic; it seems like something more specific like buildkit-pined-artifact:// or something might be more straightforward, even if this is an implementation detail/not user facing, as you said.

@tonistiigi 
> This doesn’t say anything about the transport/storage. All the current source names define that.
> But I already have a problem that I want to be able to point to a local OCI layout this way as well that would need another name.

---

Based on the above, what about `docker-image+blob` and `oci-layout+blob` for this?

Reply:
> `oci-layout+blob`

I'd definitely be interested in having this as well - making sure we can load them from disk makes this super useful for me. I don't think we should have both `oci-layout` and `docker-image` functionality merged into a new `docker-image-blob`, so the `+` sounds like a neat solution to me :tada:

Reply:
I don't feel like it's worth bikeshedding this too much: either `docker-image+blob`/`oci-layout+blob` or `docker-image-blob`/`oci-layout-blob` would work for me.

The `+` implies to me shared code / similar functionality to the base, which isn't quite true though.

Reply:
It is though, from the perspective of a user: both come from a registry/an OCI bundle.